### ISSUE TITLE
Unmaximizing with maximize.sh makes window from right frame briefly flicker in left frame

### ISSUE THREAD

**Issue by astier** [herbstluftwm/herbstluftwm#1237]
## Info

herbstluftwm 0.9.2 (2f337ab3 built on 2021-03-15)

## Problem

When maximize.sh is used two maximize two frames then the window from the right frame appears briefly in the left frame if left frame is unmaximized with maximize.sh

## Reproduce

1. Kill any compositor like picom or xcompmgr
1. Create two frames. One on the left and one on the right side of the screen.
2. Open a terminal in the left frame and another terminal-window in the right frame.
3. Generate a lot of text in the right terminal (ex. run htop)
4. Focus left frame
4. Maximize both frames by executing maximize.sh in the left terminal
5. Unmaximize both frames by executing maximize.sh in the left terminal

It can be seen how the text from the right terminal briefly can be seen in the left frame.

## Expected

The right window shouldn't be seen in the left frame when the left frame is unmaximized with maximize.sh

**Comment by t-wissmann** [comment]
In order to reproduce it I additionally have to deactivate `hide_covered_windows`.

**Comment by astier** [comment]
You are right. If `hide_covered_windows` is true then the issue disappears.

**Comment by astier** [comment]
I just discovered that the same issue also happens if the window in the right frame is unminimized with tint2 or plank.

1. Create two frames. Left frame has firefox. Right frame has terminal.
2. Focus terminal.
3. Launch plank `XDG_SESSION_TYPE=x11 dbus-launch plank`
4. Click on the terminal-icon in the plank to minimize terminal.
5. Click on the terminal-icon in the plank again to unminimize terminal.

During unminimizing the terminal briefly flickers in the left frame before it is restored in the right frame.

**Comment by astier** [comment]
I guess what I am trying to say is that maybe the bug is not in the `maximize.sh` script but in the way hlwm maps windows. Looks like it maps them before it configures them.

**Comment by astier** [comment]
It seams like hlwm maps the window from the right frame first on the position 0x0 (top left corner) with the pseudotile-size before moving it to the right frame and resizing it to the frame-size. Maybe XMapWindow is called before XConfigureWindow.
